lib/miq_expression.rb

```diff
@@ -759,7 +759,8 @@ def self.get_col_info(field, options = {})
       model = ref.klass
     end
     if col
-      result[:data_type] = col_type(model, col)
+      f = Field.new(model, [], col)
+      result[:data_type] = f.column_type
       result[:format_sub_type] = MiqReport::Formats.sub_type(col.to_sym) || result[:data_type]
       result[:virtual_column] = model.virtual_attribute?(col.to_s)
```

Comment:
@imtayadeway do you want me to call `virtual_attribute?` and `supported_by_sql?` on the field instance?

I currently have a vendetta to this whole method. (along with anything that is building a hash to define a model / column)